documentation/CAMARA-Security-Interoperability.md

```diff
@@ -0,0 +1,158 @@
+
+### Authentication Request
+
+This CAMARA document makes the login_hint parameter REQUIRED if CIBA is used in a CAMARA context.
+
+This CAMARA document clarifies the values used in login_hint in the following way:
+
+    * **_tel_**
+
+      For phone numbers. The `login_hint` must be a tel URI as defined in [RFC 3966](https://www.rfc-editor.org/info/rfc3966) for global phone numbers without visual separators in [E.164](https://www.itu.int/rec/T-REC-E.164-201011-I/en) format. For example, `tel:+34666666666`.
+
+    * **_ipport_**
+
+      For IPv4 and IPv6 addresses, that can optionally include a port. For example, `ipport:80.90.34.2:16790`, `ipport:80.90.34.2`, `ipport:[2001:db8::1]:8080` or `ipport:[2001:db8::1]`.
+
+
+The client MUST authenticate with the authorization server as described in [Client Authentication Section](#client-authentication).
+
+## Client Credentials Flow
+
+The [OAuth 2.0 Client Credentials](https://datatracker.ietf.org/doc/html/rfc6749#section-4.4) grant type is used to obtain a 2-legged Access Token that does not represent a user. This grant type can only be used when no personal user data is processed, and it is only a valid option to access the CAMARA APIs for these specific scenarios.
+
+
+### Access Token Request
+
+This CAMARA document makes the scope parameter REQUIRED for the OAuth2 Client Credentials Grant.
+
+The client MUST authenticate with the authorization server as described in [Client Authentication Section](#client-authentication).
+
+## The Scope Parameter
+
+Scope values determine the specific CAMARA services being requested by the Service Provider, subject to the SP being registered to use those services. The scope values must be documented in the API OAS files by all Camara API subprojects.
+
+How to apply the concept of purpose with scope can be found in [Section Purposes of CAMARA APIs access and user consent management](https://github.com/camaraproject/IdentityAndConsentManagement/blob/main/documentation/CAMARA-API-access-and-user-consent.md).
+
+## ID Token sub claim
+
+The subject identifier must be globally unique identifier and a case-sensitive ASCII string with a maximum length of 255. The sub value MUST not contain the MSISDN. Within Camara this is populated by a Pseudonymous Customer Reference (PCR) which is a system-facing unique identifier that links a User’s MSISDN/IP-Address to a ASP. It is generated by the OP.
+
+## Client Authentication
+
+This CAMARA document allows **one** client authentication method, private_key_jwt, as defined in OIDC
+[OIDC Client Authentication](https://openid.net/specs/openid-connect-core-1_0.html#ClientAuthentication)
+[JSON Web Token (JWT) Profile for OAuth 2.0 Client Authentication and Authorization Grants](https://www.rfc-editor.org/rfc/rfc7523.html#section-2.2)
+
+## Sender-constraint Access Token
+
+This CAMARA document RECOMMENDS sender-constraint access tokens.
```

Comment:
This has never been discussed.

Reply:
That might be true, but this document is about Camara security and sender-constraint access tokens add  considerably to Camara security. [FAPI 2.0](https://openid.bitbucket.io/fapi/fapi-2_0-security-profile.html) requires sender-constraint access tokens. TMOUS expressed interest in sender-constraint access tokens. I guess because US telcos are nearly there to implement DPoP.

So, I suggested that ICM recommends sender-constraint access tokens and require DPoP support for AZ.

Starting the discussion now

Reply:
@jpengar You are always arguing over processes but not over content. 
I agree that putting sender-constraint assertions into the document from https://github.com/camaraproject/IdentityAndConsentManagement/pull/113 would make that document harder to comprehend. It is already very long and hard to review because the differences  to the standards are not clearly marked. 

If we continue with the #113 approach, we would probaly need to repeat DPop and also add text about sender-constraint assertions.

With this short and concise document adding sender-constraint assertions is no big deal which improves Camara security incredibly. 

Of course, only if WG members agree and support sender-constraint assertions implemented using DPop.

Reply:
>@jpengar You are always arguing over processes but not over content.

@AxelNennker Ido really  hope this is the last time I have to ask you to keep the discussion out of the personal realm. And yes, as a code owner, I care about processes. I try to care about the scope of project releases, I try to care about the way we work, and I try to keep discussions separate when possible, because in my experience, and especially in my past experience in this WG, that's the best way to get results.
And as for the part where you say that I don't discuss content, please take a look at the posting history of this WG, I can guarantee you that I have discussed content a lot. So that's pretty unfair. 

>I agree that putting sender-constraint assertions into the document from https://github.com/camaraproject/IdentityAndConsentManagement/pull/113 would make that document harder to comprehend. It is already very long and hard to review because the differences to the standards are not clearly marked.

We pretty much know DT's position on the document format, these kinds of comments are starting to be quite disrespectful.

>If we continue with the https://github.com/camaraproject/IdentityAndConsentManagement/pull/113 approach, we would probaly need to repeat DPop and also add text about sender-constraint assertions.

As explained in the call and by my colleague @palmerabollo, we can actually forget about the format of the document itself (Mobile Connect profile is a word document for example). We just need to make sure that the OIDC profile we create contains all the necessary information to ensure interoperability of CAMARA implementations. That's the goal and nothing else. Having said that, I don't mind using DT's PR as a reference as long as it includes and solves all interoperability issues. And of course, if we keep topics not discussed yet for a next release, I think that is fair.

> With this short and concise document adding sender-constraint assertions is no big deal which improves Camara security incredibly.

As recommended in the last call, to add something that hasn't been discussed before, there should be consensus in the working group, right? The same applies here, and to do that we need to open this issue for discussion for the next release. I would recommend opening specific issues on github for this, and including a v0.3.0 tag to properly identify them.

> Of course, only if WG members agree and support sender-constraint assertions implemented using DPop.

Of course, indeed.

Reply:
Sorry, I did not mean to be disrespectful.

Reply:
To the topic of sender-constraint access token and PKCE requirements: I am going to remove these security improving requirements. I expect these topics are going to be put in issues and PR after this PR is merged

Reply:
removed